lib/facter/secure_boot_enabled.rb

```diff
@@ -0,0 +1,25 @@
+# _Description_
+#
+# Return true if system booted via uEFI Secure Boot
+#
+if Facter.value(:kernel).downcase == "linux" then
+  Facter.add("secure_boot_enabled") do
+    setcode do
+      if File.exist?('/sys/firmware/efi')
```

Comment:
Moving this to a confine could be useful as well for readability:

`confine { File.exist?('/sys/firmware/efi') }`

Reply:
From a work flow perspective, wouldn't that prevent the fact from being defined on non-efi systems?  With facter considering `undef == true`, that feels like it would lead to false positives....

Reply:
In `facter` `undef` is `undef` and the Puppet language treats it as `false`.

We've trended towards using `dig` for pretty much everything like `if dig($facts, 'secure_boot_enabled') { ... }` and this takes care of any weird cases.

That said, I see your point. I think just moving to the `retval` style would help in terms of understanding that this fact should either be `true` or `false` and you're right that this should not be moved to a `confine` since that would be an unknown state.